### PR TITLE
#184 Introduce a new "expand" api to request all connectors status and config info

### PR DESCRIPTION
**Pull request by harveyyue**
https://github.com/kcctl/kcctl/issues/184

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Hey @harveyyue, so how exactly will the output of `kcctl get connectors` differ with this change? Also, shouldn't the "expand" attribute be a command parameter somehow?

**Comment by harveyyue** [comment]
@gunnarmorling 

> so how exactly will the output of kcctl get connectors differ with this change? 
Expose the worker_id including ip and port to TASKS, since we deployed the connect service with cluster mode, this is help us to known the connector location.

The command `kcctl get connectors` ouput example:
`
harvey_test_topic_strategy_jdbc_sink          sink        RUNNING   0: some_ip:28083 FAILED
harvey_test_sechub_v2_source                     source   RUNNING   0: some_ip:28083 RUNNING
`

> Also, shouldn't the "expand" attribute be a command parameter somehow?
Yes, only add a "expand" api to get all connectors status/configs instead of getting these info one by one with related api.

**Comment by gunnarmorling** [comment]
Aaah, so I think I get it now. This doesn't change at all how the command
behaves to the user, it "only" implements it more efficiently by doing just
a single call to Connect, right?

That makes sense; we'd need to check the backend version though in order to
make sure we talk to a version new enough so it actually supports that new
call. Otherwise, we'd have to fall back to the existing implementation.

**Comment by gunnarmorling** [comment]
Here's another case btw. where we check the version of Kafka Connect and take action based on that: https://github.com/kcctl/kcctl/blob/main/src/main/java/org/kcctl/command/DescribeConnectorCommand.java#L74-L83.

**Comment by harveyyue** [comment]
Connect introduce this "expand" feature since 2.3.0-rc1. Yes, we need to add this version switch to invoke the corrected api.
[ConnectorsResource](https://github.com/apache/kafka/blob/2.3.0-rc1/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java)

File: src/main/java/org/kcctl/service/TaskInfo.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.kcctl.service;
+
+public class TaskInfo {
```
Comment (gunnarmorling) [inline src/main/java/org/kcctl/service/TaskInfo.java:18]: Could be a record type, given you started using them elsewhere? Same for a few other types.

**Comment by gunnarmorling** [comment]
LGTM, thanks a lot, @harveyyue! Anything else you wanna push to this PR before I merge it?

**Comment by gunnarmorling** [comment]
Applied; thx, @harveyyue!

**Pull request merged**